### PR TITLE
Chore: Add generic currency formatter without a prefix

### LINKED ISSUE #37838: Formatting Currency without abbreviation
**What happened**:
Wanted to output a currency value as the full value, without abbreviation, a currency symbol and thousand commas for readability. 

**What you expected to happen**:
To get something like $53,456.75 instead of $53.46K or $53456.75 or 53,456.75

**How to reproduce it (as minimally and precisely as possible)**:
Set a values units in the following ways:

* Currency / Dollar ($) - this gives you the abbreviated version which is no good for reporting.
* prefix:$ - this gives you an unabbreviated version but without the commas, making large numbers hard to read.
* Misc / Locale Format - this gives you the unabbreviated version with the commas but no currency sign.

None of these get the formatting required.

**Anything else we need to know?**:
I would suggest there are couple of options to fix this, either:

* Add "prefix" as a separate option for a value so it can have both a "prefix" and a unit type attached to it. In this way we could prefix it with $ and set the unit to "Misc / Locale Format".

OR 

* Add an extra option to the "Currency" unit option to allow for the abbreviation to be turned off.

I think the first of these offers the most flexibility but the second is more intuitive.

Forum post: https://community.grafana.com/t/formatting-currency-without-abbreviation/50287

**Environment**:
- Grafana version: 8.0.5
- Data source type & version: MySQL 8.0.19
- OS Grafana is installed on: Docker
- User OS & Browser: Mac / Chrome
- Grafana plugins: None

### PR DESCRIPTION
**Pull request by grutherford**
**What is this feature?**

Adding the option to not have a currency prefixed while still getting the number formatted. 

**Why do we need this feature?**

Currently, there are 30 currencies supported by Grafana. However, there are [180 currencies](https://en.wikipedia.org/wiki/List_of_circulating_currencies) in use in the world. This generic currency means you can get the benefit of the currency format without having to create a pull request for a specific currency. 

**Who is this feature for?**

- All users who use one of the 150 unsupported currencies
- Users who want to specify the prefix manually, e.g. using "Display name" to create `USD 2.4M`
- All non-technical users who can't create a PR to support their currency.

**Which issue(s) does this PR fix?**:

None

Please check that:
- [x] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [x] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: packages/grafana-data/src/valueFormats/categories.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ export const getCategories = (): ValueFormatCategory[] => [
       { name: 'Guaraní (₲)', id: 'currencyPYG', fn: currency('₲') },
       { name: 'Uruguay Peso (UYU)', id: 'currencyUYU', fn: currency('UYU') },
       { name: 'Israeli New Shekels (₪)', id: 'currencyILS', fn: currency('₪') },
+      { name: 'Generic (No prefix)', id: 'currency', fn: currency('') },
```
Comment (tskarhed) [inline packages/grafana-data/src/valueFormats/categories.ts:152]:
```suggestion
      { name: 'Generic', id: 'currency', fn: currency('') },
```
We cannot assume that currencies generally are prefixes.

**Review by tskarhed — COMMENTED** [review]
I don't think this solves https://github.com/grafana/grafana/issues/37838, as it talks more about values being abbreviated. There are changes needed to the actual formatting.

Reading through the issue, the correct format is achieved through Misc/Locale format, but in that case it lacks the currency sign. The hard part here is that sometimes you want the abbreviated version, and I am not quite sure how to approach that.

File: packages/grafana-data/src/valueFormats/categories.ts (hunk 1 above)
Comment (grutherford) [reply packages/grafana-data/src/valueFormats/categories.ts:152]: Agreed!

**Comment by grutherford** [comment]
> I don't think this solves #37838, as it talks more about values being abbreviated. There are changes needed to the actual formatting.
> 
> Reading through the issue, the correct format is achieved through Misc/Locale format, but in that case it lacks the currency sign. The hard part here is that sometimes you want the abbreviated version, and I am not quite sure how to approach that.

I agree I am not closing this PR at all, I think you're right that this PR wants full currency without abbreviation. 

Where as what i'm solving is currency formatting without a prefix, as you've rightly stated not every currency has a prefix.

I've removed #37838.

Can this be accepted as a valid improvement without a PR?

**Comment by tskarhed** [comment]
Otherwise you could pick up [this PR](https://github.com/grafana/grafana/pull/54691), which solves the issue by add `currency:financial:$` for example. It seems like the original creator didn't sign the CLA and closed it.

**Comment by grutherford** [comment]
> Otherwise you could pick up [this PR](https://github.com/grafana/grafana/pull/54691), which solves the issue by add `currency:financial:$` for example. It seems like the original creator didn't sign the CLA and closed it.

@tskarhed I'm not in a position to pick up that PR I'm afraid, sorry.

**Comment by tskarhed** [comment]
I am handing this over to @grafana/dashboards-squad

**Comment by grutherford** [comment]
@zserge it looks like this was closed without being included in 11.4.1. Have you guys decided on a different approach to resolving this issue?

**Comment by grutherford** [comment]
@tskarhed any chance this can be reconsidered?

**Pull request closed without merging**